src/Node/MacroNode.php

```diff
@@ -104,7 +104,13 @@ public function compile(Compiler $compiler)
             ->outdent()
             ->write("]);\n\n")
             ->write("\$blocks = [];\n\n")
-            ->write("ob_start(function () { return ''; });\n")
+        ;
+        if ($compiler->getEnvironment()->isDebug()) {
```

Comment:
Do we take the debug mode into account in the cache key ? If no, we would need to make that check at runtime rather than at compile time.

Reply:
I don't recall ... :/

Reply:
checked it. The Environment puts it in its option hash, so that's fine.